applications/StructuralMechanicsApplication/tests/test_linear_constraints.py

```diff
@@ -0,0 +1,219 @@
+        normal_2 = KratosMultiphysics.Array3([dy,-dx,0])/math.sqrt(dx**2+dy**2)
+        inner_prod(normal_2,normal_2)
+        master_dofs_2 = [n2.GetDof(KratosMultiphysics.DISPLACEMENT_Y)]
+        slave_dofs_2 = [n2.GetDof(KratosMultiphysics.DISPLACEMENT_X)]
+        RelationMatrix2 = KratosMultiphysics.Matrix(1,1)
+        RelationMatrix2[0,0] = -normal_2[1]/normal_2[0]
+        ConstantVector = KratosMultiphysics.Vector([0.0])
+
+        constraint_2 = KratosMultiphysics.LinearMasterSlaveConstraint(2,
+ 
+            master_dofs_2, 
+            slave_dofs_2, 
+            RelationMatrix2, 
+            ConstantVector)
+        mp.AddMasterSlaveConstraint(constraint_2)
+
+        ################# apply a slip constraint on node 4 
+        # note that SlipConstraint is practically a placeholder for the operations done just above
+        #node4 is at allowed to slide normally to normal4
+        dx = n1.X - n4.X
+        dy = n1.Y - n4.Y
+        normal_4 = KratosMultiphysics.Array3([dy,-dx,0])/math.sqrt(dx**2+dy**2)
+        n4.SetSolutionStepValue(KratosMultiphysics.NORMAL, normal_4) #artificially set the normal
+
+        constraint_4 = KratosMultiphysics.SlipConstraint(4,
+            n4,
+            KratosMultiphysics.DISPLACEMENT_X,
+            KratosMultiphysics.DISPLACEMENT_Y,
+            KratosMultiphysics.DISPLACEMENT_Z,
+            KratosMultiphysics.NORMAL)
+        mp.AddMasterSlaveConstraint(constraint_4)
+
+        #solve the problem
+        strategy = self._create_strategy(mp)
+        self._solve_with_strategy(strategy,0)
+
+        ##verify the results
+        d1 = n1.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT)
+        d2 = n2.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT)
+        d3 = n3.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT)
+        d4 = n4.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT)
+
+        self.assertAlmostEqual(d2[0]*normal_2[0] + d2[1]*normal_2[1],0.0, 15)
+        self.assertAlmostEqual(d4[0]*normal_4[0] + d4[1]*normal_4[1],0.0, 15)
+        self.assertAlmostEqual(d3[0], 0.0, 15) #symmetry condition       
+        self.assertAlmostEqual(d3[1], 2.0*d2[1], 15)
+
+        R2 = n2.GetSolutionStepValue(KratosMultiphysics.REACTION)
+        R4 = n4.GetSolutionStepValue(KratosMultiphysics.REACTION)
+
+        nR2 = inner_prod(normal_2, R2)
+        nR4 = inner_prod(normal_4, R4)
+
+        #check that Reactions are in compression
+        self.assertTrue(nR2 < 0.0)
+        self.assertTrue(nR4 < 0.0)
+        
+        #check that tangential component is zero
+        tang_2 = R2 - nR2*normal_2
+        tang_4 = R4 - nR4*normal_4
+
+        self.assertTrue(tang_2.norm_2() < 1e-12)
+        self.assertTrue(tang_4.norm_2() < 1e-12)
+
+        self.assertEqual(mp.ProcessInfo[KratosMultiphysics.NL_ITERATION_NUMBER], 5) #4 if using Residual Criteria
+
+    def __post_process(self, main_model_part):
+        from gid_output_process import GiDOutputProcess
```

Comment:
please either remove or fix, this does not work

Reply:
removed, it was a leftover